parking_permits_app/resolvers.py

```diff
@@ -96,25 +96,31 @@ def resolve_delete_parking_permit(obj, info, permit_id, customer_id):
 @mutation.field("createParkingPermit")
 @authenticate_parking_permit_token
 @convert_kwargs_to_snake_case
-def resolve_create_parking_permit(obj, info, customer_id, zone_id, registration):
+def resolve_create_parking_permit(obj, info, customer_id, zone_id, registrations):
     customer = Customer.objects.get(id=customer_id)
-
-    try:
-        permit = ParkingPermit.objects.get(
-            Q(vehicle__registration_number__iexact=registration),
-            Q(vehicle__owner=customer) | Q(vehicle__holder=customer),
-        )
-    except ObjectDoesNotExist:
-        customer_vehicles = Vehicle.objects.filter(
-            Q(owner=customer) | Q(holder=customer),
-        ).exclude(registration_number__iexact=registration)
-        permit = ParkingPermit.objects.create(
-            customer=customer,
-            parking_zone=ParkingZone.objects.get(id=zone_id),
-            primary_vehicle=len(customer_vehicles) == 0,
-            vehicle=get_mock_vehicle(customer, registration),
-        )
-    return {"success": True, "permit": resolve_prices_and_low_emission(permit)}
+    ParkingPermit.objects.filter(
+        Q(status=constants.ParkingPermitStatus.DRAFT.value),
+        Q(vehicle__owner=customer) | Q(vehicle__holder=customer),
+    ).exclude(Q(vehicle__registration_number__in=registrations)).delete()
+
+    for registration in registrations:
+        try:
+            ParkingPermit.objects.get(
+                Q(vehicle__registration_number__iexact=registration),
+                Q(vehicle__owner=customer) | Q(vehicle__holder=customer),
+            )
+        except ObjectDoesNotExist:
```

Comment:
Minor and not block the merge, but I think more specific exceptions should be preferred whenever possible, even though we only used a single model in the try block.

`except ParkingPermit.DoesNotExist:`

Reply:
Done